lib/Schema.ts

```diff
@@ -607,7 +607,26 @@ Schema.prototype.requiredCheck = async function (this: Schema, key: string, valu
 };
 
 Schema.prototype.getIndexAttributes = async function (this: Schema): Promise<{index: IndexDefinition; attribute: string}[]> {
-	return (await Promise.all(this.attributes().map(async (attribute: string) => ({"index": await this.getAttributeSettingValue("index", attribute) as IndexDefinition, attribute})))).filter((obj) => obj.index);
+	return (await Promise.all(this.attributes()
+		.map(async (attribute: string) => ({
+			"index": await this.getAttributeSettingValue("index", attribute) as IndexDefinition,
+			attribute
+		}))
+	))
+		.filter((obj) => obj.index)
+		.reduce((accumulator, currentValue) => {
+			if (Array.isArray(currentValue.index)) {
+				currentValue.index.forEach((currentIndex) => {
+					accumulator.push({
+						...currentValue,
+						"index": currentIndex
+					});
+				});
+			} else {
+				accumulator.push(currentValue);
+			}
+			return accumulator;
+		}, []);
```

Comment:
Can we remove 1 tab of indentation from these lines?

Reply:
Per https://github.com/dynamoose/dynamoose/pull/1079/checks?check_run_id=1548607295, I guess the answer is no. Lol.

I will abide by the linter here.

If you spot anything else, feel free to adjust the styling as you see fit – it's probably faster than waiting on me to address whitespace! :P